### DISCUSSION TITLE
HEVC & AVC - Output driver - Different settings

### DISCUSSION THREAD

**Discussion by JoseteLuisete** [smplayer-dev/smplayer#248] [General]
Hi! I play AVC and HEVC videos. My multimedia engine is **mpv**.
I have no problems playing videos in AVC format. With almost any configuration the result is ... perfect color and smooth playback. The problem comes when I play videos in HEVC format.

I observe 2 situations depending on whether I select one or the other value for **Output driver**.
**Result 1**: Perfect color, jumpy playback.
**Result 2**: Dull color, smooth playback.
**Drivers with result 1**: Default / gpu / vdpau / wlshm / drm
**Drivers with result 2**: xv / vaapi

The solution I have found is:
1º Use the xv driver
2º When I play HEVC videos, activate the Video equalizer (Ctrl+E) with the values:
Contrast +5, Hue -15, Saturation +25, Software equalizer Yes
Is it possible to have different settings automatically depending on whether AVC or HEVC videos are played?

My system is Manjaro Linux KDE Plasma.
My graphics card is UHD Graphics 630 integrated with the Intel processor Intel i9-9900
Greetings!

**Comment by smplayer-dev** [comment]
Can you provide a URL of a HEVC video with this problem?

**Reply by JoseteLuisete** [reply]
It happens with all my 200 MKV HEVC movies I have.

An example ...
https://jell.yfish.us/
Video numbers 1 and 3 ...
I've made 3 captures:
A) · Video AVC / h264 with Video Equalizer OFF
B) · Video HEVC / h265 with Video Equalizer OFF
C) · Video HEVC / h265 with Video Equalizer ON (Contrast +5, Hue -15, Saturation +25)
Videos A and C are almost identical in my system.

![jellyfish-3-mbps-hd-h264 - Video Equalizer OFF - 16 seconds](https://user-images.githubusercontent.com/7236449/137579862-07ea734a-c467-4f62-9b63-bdb8effc613d.png)

![jellyfish-3-mbps-hd-hevc-10bit - Video Equalizer OFF - 16 seconds](https://user-images.githubusercontent.com/7236449/137579870-4e0d04d5-3dad-4a7b-b989-d0f29d62d548.png)

![jellyfish-3-mbps-hd-hevc-10bit - Video Equalizer ON - 16 seconds](https://user-images.githubusercontent.com/7236449/137579881-3b19ff65-9611-4ef0-a1d5-7c75e19add07.png)

**Comment by JoseteLuisete** [comment]
Hello!
Am I right about this?
Any news?
Is it going to be solved?
Thanks!

**Comment by smplayer-dev** [comment]
In any case this is a bug in mpv, so you should report it to its developers. However xv and vaapi are kind of deprecated, available for compatibiliy with old systems, so I think it's unlikely they would fix it.

**Reply by JoseteLuisete** [reply]
Understood!
Which driver should I use then for smooth video playback?
These drivers, gpu, vdpau, wlshm and drm don't work properly. The playback is jumpy.
Thanks a lot in advance!
